src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureEnvironment.cs

```diff
@@ -126,6 +126,11 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
 
         public override Symbol ContainingSymbol => _topLevelMethod.ContainingSymbol;
 
+        // Closures in the same method share the same SynthesizedClosureEnvironment. We must
+        // always return true because two closures in the same method might have different
+        // AreLocalsZeroed flags.
+        public sealed override bool AreLocalsZeroed => true;
```

Comment:
>true [](start = 55, length = 4)

Shouldn't we delegate to the _topLevelMethod?

Reply:
Or perhaps to ```OriginalContainingMethodOpt ??  _topLevelMethod```? 

---
In reply to: [175912679](https://github.com/dotnet/roslyn/pull/25276#discussion_r175912679) [](ancestors = 175912679)

Reply:
Or do we always want to initialize locals for constructors and other methods that do not represent lambdas and local functions?

---
In reply to: [175915640](https://github.com/dotnet/roslyn/pull/25276#discussion_r175915640) [](ancestors = 175915640,175912679)

Reply:
I proposed we should always initialize locals for methods in synthesized closures environments that aren't directly generated from closure code (like the constructor). That's because synthesized closures can be shared. I couldn't decide if they should instead delegate up to their containing type or module, though.

Reply:
> I proposed we should always initialize locals for methods in synthesized closures environments that aren't directly generated from closure code (like the constructor). That's because synthesized closures can be shared.

Makes sense

---
In reply to: [175926195](https://github.com/dotnet/roslyn/pull/25276#discussion_r175926195) [](ancestors = 175926195)